flake.nix

```diff
@@ -4,22 +4,37 @@
 
   outputs = { self, nixpkgs }: {
     lib = {
-      # TODO handle multiple architectures.
-      evalConfig = import ./eval-config.nix { inherit (nixpkgs) lib; };
-
-      darwinSystem =
-        { modules, inputs ? { }
-        , system ? throw "darwin.lib.darwinSystem now requires 'system' to be passed explicitly"
-        , ...
-        }@args:
-        self.lib.evalConfig (args // {
-          inherit system;
-          inputs = { inherit nixpkgs; darwin = self; } // inputs;
-          modules = modules ++ [ self.darwinModules.flakeOverrides ];
-        });
+      evalConfig = import ./eval-config.nix;
+
+      darwinSystem = args@{ modules, ... }: self.lib.evalConfig (
+        { inherit (nixpkgs) lib; }
+        // nixpkgs.lib.optionalAttrs (args ? pkgs) { inherit (args.pkgs) lib; }
+        // builtins.removeAttrs args [ "system" "pkgs" "inputs" ]
+        // {
+          modules = modules
+            ++ nixpkgs.lib.optional (args ? pkgs) ({ lib, ... }: {
+              _module.args.pkgs = lib.mkForce args.pkgs;
+            })
+            # Backwards compatibility shim; TODO: warn?
+            ++ nixpkgs.lib.optional (args ? system) ({ lib, ... }: {
+              nixpkgs.system = lib.mkDefault args.system;
```

Comment:
In NixOS it is intentionally done like this to allow implicit default systems to be overridden by the configuration. However I guess in the case of the flake entry point there's never any default. I would lean towards keeping the behaviour as-is for consistency with NixOS, but I could also add logic for an error here if you would prefer; I didn't bother doing anything fancy as I'd prefer to deprecate passing `system` altogether and make it warn unconditionally at some point.

Reply:
Yeah, guess we can also do that.

Reply:
FWIW I can put in warnings now if you'd prefer, I was just trying to avoid the PR getting any bigger or the same change that fixes people's systems also warning about them :)